src/common/utils.py

```diff
@@ -225,7 +225,8 @@ def _get_instance_info_from_pricing_file(region, proxy_config, instance_type):
         log.critical(error_msg)
         raise CriticalError(error_msg)
 
-    return _get_vcpus_by_instance_type(instances, instance_type), _get_gpus_by_instance_type(instances, instance_type)
+    vcpus, memory = _get_vcpus_mem_by_instance_type(instances, instance_type)
```

Comment:
minor: I'd put some order here since it's a bit confusing having a function for vcpus and memory and one for gpu.

- the gpu one can actually be replaced by this 1 liner: `int(instances[instance_type].get("gpu", 0))`
- the vcpu can be also a 1 liner: `int(instances[instance_type]["vcpus"])`
- the memory can actually also be a 1 liner: `int(float(instances[instance_type].get("memory", 0))*1000)`. But then we need to handle memory==0 in the caller

I'd have a single try except around the 3 operations above where we fail with critical and a clear error message (as already done) in case we cannot find the instance in the json doc.